fastify.js

```diff
@@ -43,6 +52,10 @@ function build () {
   }
 
   function put (url, schema, handler) {
+    if (!handler && typeof schema === 'function') {
+      handler = schema
+      schema = {}
+    }
```

Comment:
we can probably put this into a little helper, something like `route(fill({ method: 'PUT', url }, schema, handler))`.
Or `_route({ method: 'PUT', url }, schema, handler)`, which contains this logic.

The important piece is that it should not be applied to the `options`, but to the function arguments.